pandaserver/dataservice/setupper_atlas_plugin.py

```diff
@@ -0,0 +1,2226 @@
+                        # doesn't exist in DQ2
+                        if status == -1:
+                            missing_ds[dataset] = f"DS:{dataset} not found in DDM"
+                        else:
+                            missing_ds[dataset] = out
+                    else:
+                        # make map (key: LFN w/o attemptNr, value: LFN with attemptNr)
+                        items = out
+                        try:
+                            # loop over all files
+                            for tmp_lfn in items:
+                                vals = items[tmp_lfn]
+                                val_map[tmp_lfn] = vals
+                                gen_lfn = re.sub("\.\d+$", "", tmp_lfn)
+                                if gen_lfn in lfn_map[dataset]:
+                                    # get attemptNr
+                                    new_att_nr = 0
+                                    new_mat = re.search("\.(\d+)$", tmp_lfn)
+                                    if new_mat is not None:
+                                        new_att_nr = int(new_mat.group(1))
+                                    old_att_nr = 0
+                                    old_mat = re.search("\.(\d+)$", lfn_map[dataset][gen_lfn])
+                                    if old_mat is not None:
+                                        old_att_nr = int(old_mat.group(1))
+                                    # compare
+                                    if new_att_nr > old_att_nr:
+                                        lfn_map[dataset][gen_lfn] = tmp_lfn
+                                else:
+                                    lfn_map[dataset][gen_lfn] = tmp_lfn
+                                # mapping from LFN to DS
+                                lfn_ds_map[lfn_map[dataset][gen_lfn]] = dataset
+                        except Exception:
+                            prod_error[dataset] = f"could not convert HTTP-res to map for prodDBlock {dataset}"
+                            self.logger.error(prod_error[dataset])
+                            self.logger.error(out)
+                    # get replica locations
+                    if (self.only_ta or job.prodSourceLabel in ["managed", "test"]) and prod_error[dataset] == "" and dataset not in replica_map:
+                        if dataset.endswith("/"):
+                            status, out = self.get_list_dataset_replicas_in_container(dataset, True)
+                        else:
+                            status, out = self.get_list_dataset_replicas(dataset)
+                        if not status:
+                            prod_error[dataset] = f"could not get locations for {dataset}"
+                            self.logger.error(prod_error[dataset])
+                            self.logger.error(out)
+                        else:
+                            replica_map[dataset] = out
+                            # append except DBR
```

Comment:
what is DBR (DB release? I've heard the word, but I don't really know what it is). what is ddo? is this still relevant?

Reply:
probably obsolete